lib/liquid/template.rb

```diff
@@ -64,11 +62,11 @@ class << self
       attr_writer :taint_mode
 
       def file_system
-        @@file_system
+        Thread.current[:liquid_file_system] ||= BlankFileSystem.new
```

Comment:
Well, it's `Thread.current`, how would that be a race condition? I think it's a pretty common pattern to ensure thread-safety, no?